Add note about delegate_facts to run_once (#58241)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_delegation.rst

.. note::
    Any conditional (i.e `when:`) will use the variables of the 'first host' to decide if the task runs or not, no other hosts will be tested.

.. note::

>     If facts should *not* be set for all hosts `delegate_facts: True` also needs to be specified for the specific task or block.

Reviewer: it 'should'  by default, better phrasing is `if you want to avoid the default behaviour of setting the fact for all hosts...`

>     If you want to avoid the default behaviour of setting the fact for all hosts `delegate_facts: True` also needs to be specified for the specific task or block.

Reviewer 2: ```suggestion
    If you want to avoid the default behaviour of setting the fact for all hosts, set `delegate_facts: True` for the specific task or block.
```
